Plugins/Flow.Launcher.Plugin.PluginsManager/PluginsManager.cs

```diff
@@ -292,10 +293,87 @@ await Http.DownloadAsync(x.PluginNewUserPlugin.UrlDownload, downloadToFilePath)
                         ContextData =
                             new UserPlugin
                             {
-                                Website = x.PluginNewUserPlugin.Website, UrlSourceCode = x.PluginNewUserPlugin.UrlSourceCode
+                                Website = x.PluginNewUserPlugin.Website,
+                                UrlSourceCode = x.PluginNewUserPlugin.UrlSourceCode
                             }
                     });
 
+            if (resultsForUpdate.Count() > 1)
+            {
+                var updateAllResult = new Result
+                {
+                    Title = Context.API.GetTranslation("plugin_pluginsmanager_update_all_title"),
+                    SubTitle = Context.API.GetTranslation("plugin_pluginsmanager_update_all_subtitle"),
+                    IcoPath = icoPath,
+                    AsyncAction = async e =>
+                    {
+                        string message;
+                        if (Settings.AutoRestartAfterChanging)
+                        {
+                            message = string.Format(Context.API.GetTranslation("plugin_pluginsmanager_update_all_prompt"),
+                                                    resultsForUpdate.Count(), Environment.NewLine);
+                        }
+                        else
+                        {
+                            message = string.Format(Context.API.GetTranslation("plugin_pluginsmanager_update_all_prompt_no_restart"),
+                                                    resultsForUpdate.Count());
+                        }
+
+                        if (MessageBox.Show(message,
+                            Context.API.GetTranslation("plugin_pluginsmanager_update_title"),
+                            MessageBoxButton.YesNo) == MessageBoxResult.No)
+                        {
+                            return false;
+                        }
+
+                        await Task.WhenAll(resultsForUpdate.Select(async plugin =>
```

Comment:
Sorry you may understand my idea wrongly. Parallel.ForeachAsync is still probably the best to use.